libraries/report.rb

```diff
@@ -28,13 +28,32 @@ class ComplianceReport < Chef::Resource
       blob[:reports] = reports
       blob[:profiles] = ownermap
 
-      Chef::Config[:verify_api_cert] = false
-      Chef::Config[:ssl_verify_mode] = :verify_none
+      # resolve owner
+      o = return_or_guess_owner
 
-      url = construct_url(::File.join('/organizations', org, 'inspec'))
-      # Chef::Log.debug "url: #{url}"
-      rest = Chef::ServerAPI.new(url, Chef::Config)
-      rest.post(url, blob)
+      if token
+        if direct
```

Comment:
I've changed it to `variant`. Note that this is not what triggers the endpoint, but only the _way it is used in compliance_: should it use a chef-server connector token and setup a proper chef-source for the node + env (`:chef`), or is it merely a non-chef-connector token (`:compliance`) and the stuff should be owned by the owner that is given via the other property.

If it's using the chef-gate API or directly calls out to a compliance API endpoint is determined by the presence of the `token` property.